lib/RequestShortener.js

```diff
@@ -5,39 +5,55 @@
 "use strict";
 
 const path = require("path");
+const NORMALIZE_SLASH_DIRECTION_REGEXP = /\\/g;
+const PATH_CHARS_REGEXP = /[-[\]{}()*+?.,\\^$|#\s]/g;
+const SEPERATOR_REGEXP = /[\/\\]$/;
+const FRONT_OR_BACK_BANG_REGEXP = /^!|!$/;
```

Comment:
is this supposed to be `/^!|!$/g` ? 
Otherwise it will only remove the first "!" so "!foo!" will turn into "foo!"